rest/codegen/server.go

```diff
@@ -1,11 +1,288 @@
 package codegen
 
 import (
+	"fmt"
+	"text/template"
+
 	"goa.design/goa.v2/codegen"
+	goadesign "goa.design/goa.v2/design"
 	"goa.design/goa.v2/rest/design"
+	"net/http"
 )
 
+//todo: in encode headers and mappings
+//todo: in decode payload, query and paths
+
+const serverT = `
+{{define "handlers_struct" -}}
+// {{.Name}}Handlers lists the {{.Name}} service endpoint HTTP handlers.
+type {{.Name}}Handlers struct {
+	{{- range .Actions}}
+	{{goify .Name true}} http.Handler
+	{{- end}}
+}
+{{end}}
+
+{{define "handlers_constructor" -}}
+// New{{.Name}}Handlers instantiates HTTP handlers for all the {{.Name}} service endpoints.
+func New{{.Name}}Handlers(
+	e *endpoints.{{.Name}},
+	dec rest.RequestDecoderFunc,
+	enc rest.ResponseEncoderFunc,
+	logger goa.Logger,
+) *{{.Name}}Handlers {
+	return &{{.Name}}Handlers{
+		{{- range .Actions}}
+		{{.Name}}: NewCreate{{$.Name}}Handler(e.{{.Name}}, dec, enc, logger),
```

Comment:
`Create` shouldn't be hard-coded. Same comment as above (and throughout the rest): compute the name of the handler constructors in the writer code so that it doesn't have to be done here. The input data structures could have a `Handlers` field with is a slice of `HandlerData` data structures which would define e.g. a `Constructor` field that contains the constructor name so that the template would simply do `{{ .Name }}: {{ .Constructor}}(e.{{ .EndpointName }}, ...`.